CMakeLists.txt

```diff
@@ -39,7 +40,7 @@ endif()
 #
 # tests
 #
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
+if((${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME} OR ${PROJECT_NAME}_BUILD_TESTING) AND BUILD_TESTING)
```

Comment:
not sure about the `... AND BUILD_TESTING` here, if this is required, we probably will also trigger building tests in our external requirements like libcurl and maybe others

Reply:
This line is the result of the discussion here: https://github.com/EVerest/EVerest/issues/129 and and in the `WG CI/CD & Testing`.
I understand your concern, could be handled by using the `options` field in `dependencies.yaml` (setting `BUILD_TESTING=OFF`)

Reply:
Has the discussion, you mentioned, taken into account, that requiring `BUILD_TESTING` would trigger unit tests in the dependencies?
Concerning the setting `BUILD_TESTING` in the options field - I'm not sure if this will work - because it is an option with global scope.  So it can be only on or off, but not on for one project and off for another one.